src/test/java/com/redislabs/redisgraph/RedisGraphAPITest.java

```diff
@@ -973,4 +989,127 @@ public void testCachedExecution() {
+                + "nullValue=Property{name='nullValue', value=null}, "
+                + "age=Property{name='age', value=32}}",expectedNode.getPropertyMap().toString());
+
+        Assert.assertEquals(
+                "Node{labels=[person], id=0, "
+                        + "propertyMap={name=Property{name='name', value=roi}, "
+                        + "boolValue=Property{name='boolValue', value=true}, "
+                        + "doubleValue=Property{name='doubleValue', value=3.14}, "
+                        + "nullValue=Property{name='nullValue', value=null}, "
+                        + "age=Property{name='age', value=32}}}", expectedNode.toString());
+
+        Map<String, Object> params = new HashMap<>();
+        params.put("name", name);
+        params.put("age", age);
+        params.put("boolValue", boolValue);
+        params.put("doubleValue", doubleValue);
+
+        Assert.assertNotNull(api.query("social", "CREATE (:person{name:$name,age:$age, doubleValue:$doubleValue, boolValue:$boolValue, nullValue:null})", params));
+        Assert.assertNotNull(api.query("social", "CREATE (:person{name:'amit',age:30})"));
+        Assert.assertNotNull(api.query("social", "MATCH (a:person), (b:person) WHERE (a.name = 'roi' AND b.name='amit')  " +
+                "CREATE (a)-[:knows{place:'TLV', since:2000,doubleValue:3.14, boolValue:false, nullValue:null}]->(b)"));
+
+
+
+        ResultSet resultSet = api.query("social", "MATCH (a:person)-[r:knows]->(b:person) RETURN a,r, " +
+                "a.name, a.age, a.doubleValue, a.boolValue, a.nullValue, " +
+                "r.place, r.since, r.doubleValue, r.boolValue, r.nullValue");
+        Assert.assertNotNull(resultSet);
+        //New ResultSet.getResults() Method test
+        Assert.assertNotNull(resultSet.getResults());
+        Assert.assertEquals(1,resultSet.getResults().size());
+
+        Assert.assertFalse(resultSet.getResults().isEmpty());
+        Record record = resultSet.next();
+        Assert.assertFalse(resultSet.hasNext());
+
+
+
+        Node node = record.getValue(0);
+        Node node2 = resultSet.getResults().get(0).getValue(0);
+        Assert.assertNotNull(node);
+        Assert.assertNotNull(node2);
+        Assert.assertEquals(node,node2);
+
+        Assert.assertEquals(expectedNode, node2);
+
+        node = resultSet.getResults().get(0).getValue("a");
+        Assert.assertEquals(expectedNode, node);
+
+        Edge edge = resultSet.getResults().get(0).getValue(1);
+        Assert.assertNotNull(edge);
+        Assert.assertEquals(expectedEdge, edge);
+
+        edge = resultSet.getResults().get(0).getValue("r");
+        Assert.assertEquals(expectedEdge, edge);
+
+        Assert.assertEquals(Arrays.asList("a", "r", "a.name", "a.age", "a.doubleValue", "a.boolValue", "a.nullValue",
+                "r.place", "r.since", "r.doubleValue", "r.boolValue", "r.nullValue"), resultSet.getResults().get(0).keys());
+        //Serialization tests
```

Comment:
can you add a scenario of serialization and deserialization?
resultsSet->JSON->resultSet
compare the original resultSet to the generated resultSet